v6.1.0 (#11025)
GoogleChrome/lighthouse: changelog.md

<a name="6.1.0"></a>
# 6.1.0 (2020-06-25)
[Full Changelog](https://github.com/GoogleChrome/lighthouse/compare/v6.0.0...v6.1.0)

We expect this release to ship in the DevTools of [Chrome 85](https://chromiumdash.appspot.com/schedule).

## New Contributors

Thanks to our new contributors 👽🐷🐰🐯🐻! 
Loftie Ellis <lpellis@gmail.com>
Marvin Frachet <marvin.frachet@gmail.com>
Matt Hobbs <matthew.hobbs@digital.cabinet-office.gov.uk>
Peter Marshall <p.s.marshall0@gmail.com>

> Umar Hansa <umar.hansa@gmail.com>

Reviewer: I believe he had commits land waaaaaay long ago in 1.x :) https://github.com/GoogleChrome/lighthouse/commits?author=umaar

> Loftie Ellis <lpellis@gmail.com>
> Marvin Frachet <marvin.frachet@gmail.com>
> Matt Hobbs <matthew.hobbs@digital.cabinet-office.gov.uk>
> Peter Marshall <p.s.marshall0@gmail.com>
> Umar Hansa <umar.hansa@gmail.com>

Reviewer: we should really try to map these back to github login, I guess we could replace it with a github API lookup on the commit itself?

I can look into that

for this release just manual?

```suggestion
Loftie Ellis @lpellis
Marvin Frachet @mfrachet 
Matt Hobbs @Nooshu 
Peter Marshall @psmarshall 
```

Author: yes please an ty so much for doing this bit for me :)

ref #9857

## Notable Changes

> ~~ TODO: Move notable changes here ~~

Reviewer: not sure we have major ones, 6.2 will be juicy with 'em though :)

Author: I added two things here.

* readme: update integrations section ([#10826](https://github.com/GoogleChrome/lighthouse/pull/10826))
* readme: update PageSpeed Green description ([#10818](https://github.com/GoogleChrome/lighthouse/pull/10818))

## Tests

* move proto roundtrip json to .tmp/ ([#10995](https://github.com/GoogleChrome/lighthouse/pull/10995))
* add heading key tests ([#10746](https://github.com/GoogleChrome/lighthouse/pull/10746))
* run ToT and stable Chrome for smoke tests in github workflow ([#10989](https://github.com/GoogleChrome/lighthouse/pull/10989))
* legacy-javascript: exit code 1 on failure ([#10946](https://github.com/GoogleChrome/lighthouse/pull/10946))
* smoke: use --debug in github action ([#10919](https://github.com/GoogleChrome/lighthouse/pull/10919))
* smokehouse: do not assert on flaky node path ([#10827](https://github.com/GoogleChrome/lighthouse/pull/10827))

## Misc

* fix merge error for subHeading -> subItemsHeading rename ([#10983](https://github.com/GoogleChrome/lighthouse/pull/10983))
* rename subHeading to subItemsHeading ([#10979](https://github.com/GoogleChrome/lighthouse/pull/10979))

> * viewer: expose LHR as __LIGHTHOUSE_JSON__ ([#10879](https://github.com/GoogleChrome/lighthouse/pull/10879))

Reviewer: ```suggestion
* viewer: expose LHR as `__LIGHTHOUSE_JSON__` ([#10879](https://github.com/GoogleChrome/lighthouse/pull/10879))
```

<a name="6.1.0"></a>
# 6.1.0 (2020-06-25)
[Full Changelog](https://github.com/GoogleChrome/lighthouse/compare/v6.0.0...v6.1.0)

We expect this release to ship in the DevTools of [Chrome 85](https://chromiumdash.appspot.com/schedule).

> 

Author: should we also be saying "we expect this release to ship in PageSpeed Insights in a week"?

Author: cc @exterkamp

Reviewer 2: I would say "Within 2 weeks" just to give us some runway if we end up needing to debug something/have weird issues.

## New Contributors

Thanks to our new contributors 👽🐷🐰🐯🐻! 
Loftie Ellis @lpellis
Marvin Frachet @mfrachet 
Matt Hobbs @Nooshu 
Peter Marshall @psmarshall 

## Notable Changes

> * If a page has publicly accessible JavaScript source maps, Lighthouse will collect them to enhance the `unused-javascript` audit. If future versions of Lighthouse, source maps will be used for entirely new audits.

Reviewer: Wait what happened to the whole "we should make a big todo about source maps all together"? :)

* The report now uses `KiB` instead of `KB`. This is simply a label change–the value was and still is equal to `1024 bytes`.

## New Audits

> * long-tasks: diagnostic audit for surfacing top long tasks ([#10736](https://github.com/GoogleChrome/lighthouse/pull/10736))

Reviewer 3: ```suggestion
* long-tasks: a new performance diagnostic that shows the longest main-thread-blocking tasks during load ([#10736](https://github.com/GoogleChrome/lighthouse/pull/10736))
```

> * crawlable-anchors: add an anchor-href audit for seo ([#10662](https://github.com/GoogleChrome/lighthouse/pull/10662))

Reviewer 3: ```suggestion
* crawlable-anchors: a new SEO audit that checks that anchors link to resolvable URLs ([#10662](https://github.com/GoogleChrome/lighthouse/pull/10662))
```

## Core

> * add source-maps gatherer to default config ([#10990](https://github.com/GoogleChrome/lighthouse/pull/10990))

Reviewer 3: also move PR# to notable change?

* unused-javascript: increase threshold to 20KiB ([#10906](https://github.com/GoogleChrome/lighthouse/pull/10906))
* layout-shift-elements: surface CLS contribution per shifted element ([#10968](https://github.com/GoogleChrome/lighthouse/pull/10968))
* emulation: bump chrome versions ([#10787](https://github.com/GoogleChrome/lighthouse/pull/10787))
* image-size-responsive: quantize DPRs ([#10801](https://github.com/GoogleChrome/lighthouse/pull/10801))
* long-tasks: add startTime property ([#10942](https://github.com/GoogleChrome/lighthouse/pull/10942))
* improve resilience of nodeId-dependent gatherers ([#10877](https://github.com/GoogleChrome/lighthouse/pull/10877))
* median-run: add computeMedianRun to lib ([#10859](https://github.com/GoogleChrome/lighthouse/pull/10859))
* preload: ignore cross-frame requests ([#10847](https://github.com/GoogleChrome/lighthouse/pull/10847))
* new inspector issues gatherer for Audit.IssueAdded events ([#10664](https://github.com/GoogleChrome/lighthouse/pull/10664))

> * renderer: fix null Util.i18n in PSI renderer ([#10822](https://github.com/GoogleChrome/lighthouse/pull/10822))

Reviewer 3: maybe move to Report

## Experimental

Features hidden behind the `--preset=experimental` flag.

* legacy-javascript: reduce polyfills, fix core-js import in test ([#10937](https://github.com/GoogleChrome/lighthouse/pull/10937))
* legacy-javascript: use prescriptive language in title ([#10850](https://github.com/GoogleChrome/lighthouse/pull/10850))
* legacy-javascript: fix core-js 3 detection ([#10852](https://github.com/GoogleChrome/lighthouse/pull/10852))
* legacy-javascript: use third-party-web for scoring ([#10849](https://github.com/GoogleChrome/lighthouse/pull/10849))
* duplicated-javascript: display transfer size ([#10701](https://github.com/GoogleChrome/lighthouse/pull/10701))

## Deps

* axe-core: upgrade to 3.5.5 ([#10986](https://github.com/GoogleChrome/lighthouse/pull/10986))
* upgrade chrome-launcher@0.13.3 ([#10911](https://github.com/GoogleChrome/lighthouse/pull/10911))
* snyk: update snyk snapshot ([#10840](https://github.com/GoogleChrome/lighthouse/pull/10840), [#10940](https://github.com/GoogleChrome/lighthouse/pull/10940), [#10980](https://github.com/GoogleChrome/lighthouse/pull/10980), [#11010](https://github.com/GoogleChrome/lighthouse/pull/11010))

> * remove bundlesize ([#10999](https://github.com/GoogleChrome/lighthouse/pull/10999))

Author: should these two be in Tests?

* update to lhci 0.4.0 ([#10828](https://github.com/GoogleChrome/lighthouse/pull/10828))

## Report

> * use KiB instead of KB ([#10870](https://github.com/GoogleChrome/lighthouse/pull/10870))

Reviewer 3: remove and add PR# to end of Notable Changes entry

(at least I personally find it confusing to have differently worded duplicates in changelogs and it's not clear if they're the same thing or related changes)

* metrics: use css grid so metrics are aligned ([#10789](https://github.com/GoogleChrome/lighthouse/pull/10789))
* don't dim disclaimer anchor links ([#10981](https://github.com/GoogleChrome/lighthouse/pull/10981))
* use acronyms and round metrics for shorter calc url ([#10954](https://github.com/GoogleChrome/lighthouse/pull/10954))
* update link for budgets audit ([#10944](https://github.com/GoogleChrome/lighthouse/pull/10944))
* add trailing slash to web.dev links ([#10967](https://github.com/GoogleChrome/lighthouse/pull/10967))
* fix the width of the 3-dots menu in topbar ([#10855](https://github.com/GoogleChrome/lighthouse/pull/10855))

> * fix zebra styling, sub item rows for 3p filter ([#10978](https://github.com/GoogleChrome/lighthouse/pull/10978))

Reviewer 3: combine with #10867?

* updated method signature typing to remove focusevent cast ([#10858](https://github.com/GoogleChrome/lighthouse/pull/10858))

## Clients

> * psi: show disclaimer and calclink ([#10936](https://github.com/GoogleChrome/lighthouse/pull/10936))

Reviewer 3: probably fine to move to Report rather than keep Clients just for the one (or two) things that are really about the PSI report anyways

## I18n

> * adjust LCP element description ([#11018](https://github.com/GoogleChrome/lighthouse/pull/11018))

Author: report??

Author: cc @exterkamp fyi I think we want to keep `i18n:` to just things that affect the i18n system, not that touch any string

Reviewer 2: Makes sense. Been awhile since I've used these tags 😟

Reviewer: I think that was actually my bad. I think it was `core(i18n):` and I was like "don't we have an i18n separate scope? and changed it before merge".

I kinda stand by the fact that this was an adjustment raised by i18n clarity purposes, but agreed generally strings shouldn't be `i18n` scope. Maybe `core(lcp): adjust element description` would've been best?

report feels weird since it has nothing to do with HTML report, it applies to all LHR uses

Reviewer 3: > report feels weird since it has nothing to do with HTML report, it applies to all LHR uses

yeah, it feels a little weird but I usually put these in core

<a name="6.1.0"></a>
# 6.1.0 (2020-06-25)
[Full Changelog](https://github.com/GoogleChrome/lighthouse/compare/v6.0.0...v6.1.0)

We expect this release to ship to DevTools in [Chrome 85](https://chromiumdash.appspot.com/schedule), and to PageSpeed Insights within 2 weeks.

## New Contributors

Thanks to our new contributors 👽🐷🐰🐯🐻! 
Loftie Ellis @lpellis
Marvin Frachet @mfrachet 
Matt Hobbs @Nooshu 

> Peter Marshall @psmarshall 

Reviewer 3: formatting on these doesn't look so great

## Notable Changes

> * If a page has publicly-accessible JavaScript source maps, Lighthouse will collect them to enhance the `unused-javascript` audit. In future versions of Lighthouse, source maps will be used for entirely new audits [#10990](https://github.com/GoogleChrome/lighthouse/pull/10990)

Reviewer 3: ```suggestion
* If a page has publicly-accessible JavaScript source maps, Lighthouse will collect them to enhance the `unused-javascript` audit. In future versions of Lighthouse, source maps will be used for entirely new audits ([#10990](https://github.com/GoogleChrome/lighthouse/pull/10990)).
```

> * The report now uses `KiB` instead of `KB`. This is simply a label change; the value was and still is equal to `1024 bytes` [#10870](https://github.com/GoogleChrome/lighthouse/pull/10870)

Reviewer 3: ```suggestion
* The report now uses `KiB` instead of `KB`. This is simply a label change; the value was and still is equal to `1024 bytes` ([#10870](https://github.com/GoogleChrome/lighthouse/pull/10870)).
```

> <img src="https://i.imgur.com/1gSdPw5.png" alt="Unused JavaScript with source maps" width="800px">

Reviewer 3: should be with the second bullet point?

Reviewer: I thought it was a nice showcase of both at the same time :)

Author: it shows both things tho

Reviewer 3: > I thought it was a nice showcase of both at the same time :)

> it shows both things tho

oh, haha, fair enough
